File: LeetCode_Solutions/876-hand-of-straights/hand-of-straights.java
class Solution {
    public boolean isNStraightHand(int[] hand, int groupSize) {
        Map<Integer,Integer>mp=new HashMap<>();
        for(int card:hand){
            mp.put(card,mp.getOrDefault(card,0)+1);
        }
        Arrays.sort(hand);
        for(int card:hand){
            if(mp.containsKey(card)){
                for(int i=card;i<card+groupSize;i++){
                    if(!mp.containsKey(i)){
                        return false;
                    }
                    mp.put(i,mp.get(i)-1);
                    if(mp.get(i)==0){
                        mp.remove(i);
                    }
                }
            }
        }
            return true;
    }
}

File: LeetCode_Solutions/876-hand-of-straights/hand-of-straights.py
class Solution:
  def isNStraightHand(self, hand: List[int], groupSize: int) -> bool:
    count = collections.Counter(hand)
    #print(sorted(count))
    hand.sort()
    for card in hand:
        if card in count:
            for i in range(card,card+groupSize):
                if not i in count:
                    return False
                count[i]-=1
                if count[i]==0:
                    del count[i]
    return True